internal/resource/url.go

```diff
@@ -554,6 +563,61 @@ func (f *Fetcher) fetchFromS3WithCreds(ctx context.Context, dest s3target, input
 	return nil
 }
 
+func (f *Fetcher) fetchFromAzureBlob(u url.URL, dest io.Writer, opts FetchOptions) error {
+	// Read about NewDefaultAzureCredential https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#DefaultAzureCredential
+	// DefaultAzureCredential is a default credential chain for applications deployed to azure.
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		f.Logger.Debug("failed to obtain Azure credential: %v", err)
+		return fmt.Errorf("failed to obtain Azure credential: %w", err)
+	}
+
+	// Create a context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Breakdown the URL into storage account, container, and file
+	storageAccount := fmt.Sprintf("%s://%s/", u.Scheme, u.Host)
+	pathSegments := strings.Split(strings.Trim(u.Path, "/"), "/")
+	if len(pathSegments) < 2 {
+		f.Logger.Debug("invalid URL path: %s", u.Path)
+		return fmt.Errorf("invalid URL path: %s", u.Path)
+	}
+	container := pathSegments[0]
+	file := pathSegments[1]
+
+	// Create Azure Blob Storage client
+	storageClient, err := azblob.NewClient(storageAccount, cred, nil)
+	if err != nil {
+		f.Logger.Debug("failed to create azblob client: %v", err)
+		return fmt.Errorf("failed to create azblob client: %w", err)
+	}
+
+	// Download the blob with retry logic
```

Comment:
Hum, shouldn't the retry happen as part of the wider retry loop in Ignition?

Reply:
Ah, yes that makes sense;  I did not consider that at all. Will drop the retry logic.